core/storage/email/gae_models_test.py

```diff
@@ -41,6 +41,27 @@ def mock_generate_hash(
             email_models.SentEmailModel, '_generate_hash',
             types.MethodType(mock_generate_hash, email_models.SentEmailModel))
 
+    def test_duplicate_mails_detected(self):
+        email_models.SentEmailModel.create(
+            'recipient_id', 'recipient@email.com', 'sender_id',
+            'sender@email.com', feconf.EMAIL_INTENT_SIGNUP,
+            'Email Subject', 'Email Body', datetime.datetime.utcnow())
+
+        email_models.SentEmailModel.create(
+            'recipient_id', 'recipient@email.com', 'sender_id',
+            'sender@email.com', feconf.EMAIL_INTENT_SIGNUP,
+            'Email Subject', 'Email Body', datetime.datetime.utcnow())
+
+        duplicated = email_models.SentEmailModel.check_duplicate_message(
+            'recipient_id', 'Email Subject', 'Email Body')
+
+        self.assertEqual(duplicated, True)
+
+        not_duplicated = email_models.SentEmailModel.check_duplicate_message(
```

Comment:
I recommend splitting this into its own test. By testing just a single behavior per test, you help avoid situations where tests fail for unrelated reasons (e.g. consider that there could be scenarios where check_duplicate_message incorrectly returns false, but the true case still works--seeing this distinction in behavior when one of the tests passes and the other fails may help the developer investigate the root cause of the regression).

Reply:
Didn't saw that one, fixed.
All done @BenHenning